### DISCUSSION TITLE
Add Delay For ImGui

### DISCUSSION THREAD

**Discussion by sFIsAnExpert** [ocornut/imgui#6847] [New Users Build/Link/Run issues ONLY!]
What is the proper way of adding a delay in ImGui?

Say for instance in my ImGui Window I want a delay similar to WinAPI Sleep() Function.

I want to output text to ImGui Window but only after the delay of 5 seconds is over, for instance -

```
if (ImGui::Delay(&iDelay)) {
	//ImGui::SetDelay(&iDelay, 5); For restarting the delay if needed
	//do whatever
}
```

I have implemented something similar to this which is compatible with ImGui but I am just wondering why is there not one currently and will we have it in the future?

**Comment by nicolasnoble** [comment]
ImGui is an immediate mode API. By design, everything is immediate, and there would be no delaying possible within the API itself.

What you could be doing is hold on to an external delay mechanism yourself, similar to the API you're describing. Something along the lines of:

```c++
if (ImGui::Button("Some button")) {
    m_deadline = std::chrono::steady_clock::now() + 5ms;
}

if (std::chrono::steady_clock::now() >= m_deadline) {
    m_deadline = std::chrono::steady_clock::now() + 5ms;
    // do whatever
}
```

**Reply by sFIsAnExpert** [reply]
I never thought about doing it that way, very neat and useful. I had my own alternative to delay but I think this one is a lot more legible. Thanks
